htdocs/core/class/html.form.class.php

```diff
@@ -1553,6 +1553,12 @@ public function select_thirdparty_list($selected = '', $htmlname = 'socid', $fil
 		if (!$user->hasRight('societe', 'client', 'voir')) {
 			$sql .= ", " . $this->db->prefix() . "societe_commerciaux as sc";
 		}
+		if (isModEnabled('category')) {
+			$sql .= " LEFT JOIN " . $this->db->prefix() . "categorie_societe as cats ON cats.fk_soc = s.rowid";
```

Comment:
Filter on category must now no more use a left join.
Filter must be implemented by adding code in where and only in WHERE.
You can take a look how the filter is implemented on project list into project/list.php, you should use AND EXISTS or NOT EXISTS depending if you need a filter on a category or "no link to category".

Reply:
I get it ! 
I have to use hook ? Or can I suggest a new param in function select_thirdparty_list() ? But it's strange to add a parameter dependent on the activation of another module... What do you think ?

Reply:
No, ne need of hook.
Just a different writing of the sql to use NOT EXISTS in the where instead of JOIN LEFT...